File: skill_sdk/tests/fixtures/cpp-project/src/core/Handler.cpp
#include "Handler.h"
#include "Helper.h"

#include <iostream>
#include <stdexcept>

Handler::Handler(std::shared_ptr<DataProcessor> processor)
    : processor_(std::move(processor)) {}

std::string Handler::ProcessRequest(const std::string& payload) {
    std::cout << "Processing request: " << payload << std::endl;

    auto helper = NewHelper(processor_);
    auto result = helper.HandleRequest(payload);
    return result;
}

bool Handler::HealthCheck() const {
    return processor_ != nullptr;
}

File: skill_sdk/tests/fixtures/rust-project/src/core/handler.rs
use crate::core::data_processor::DataProcessor;
use crate::core::helper::new_helper;

/// Handler processes incoming API requests.
///
/// Cross-file: process_request calls new_helper → handle_request.
/// goToDefinition on NewHelper from handler call resolves to helper.rs.
pub struct Handler {
    processor: Box<dyn DataProcessor>,
}

impl Handler {
    pub fn new(processor: Box<dyn DataProcessor>) -> Self {
        Self { processor }
    }

    /// ProcessRequest handles an API request end-to-end.
    /// Calls new_helper then handle_request.
    pub fn process_request(&self, payload: &str) -> String {
        println!("Processing request: {}", payload);

        let helper = new_helper(&*self.processor);
        helper.handle_request(payload)
    }

    /// HealthCheck reports the handler health status.
    pub fn health_check(&self) -> bool {
        true
    }
}
